raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -29,9 +29,15 @@ contract TokenNetwork is Utils {
     mapping (bytes32 => Channel) public channels;
 
     // We keep the unlock data in a separate mapping to allow channel data structures to be
-    // removed when settling. If there are locked transfers, we need to store data needed to
-    // unlock them at a later time. Key is the channel_identifier.
-    mapping(bytes32 => UnlockData) channel_identifier_to_unlock_data;
+    // removed when settling uncooperatively. If there are locked pending transfers, we need to
+    // store data needed to unlock them at a later time.
+    // The key is `keccak256(channel_identifier + participant_address + locksroot)`
+    // The value is the total amount of tokens locked in the pending transfers corresponding to
+    // the `locksroot`, that were sent by `participant_address` to his partner from the channel
+    // represented by the `channel_identifier`.
+    // Note that the assumption is that no two locksroots can be the same due to having different
+    // values for the secrethash of each lock, combined with different `expiration` values.
+    mapping(bytes32 => uint256) locksroot_identifier_to_locked_amount;
 
```

Comment:
Yes. 
The idea was to be able to close & settle the channel & remove all the channel data that is not needed anymore (deleting storage frees up gas and the `settleChannel` transaction costs less). 
The fact that you can also leave these pending transfer tokens in the contract and get them at a later point is an additional perk - in case the tokens worth less than the cost of the transaction.

So, `we wanna allow the reopening of channels with locked tokens` is correct.